src/rgw/rgw_asio_frontend.cc

```diff
@@ -1129,17 +1163,36 @@ void AsioFrontend::unpause()
 
   // start accepting connections again
   for (auto& l : listeners) {
-    l.acceptor.async_accept(l.socket,
-                            [this, &l] (boost::system::error_code ec) {
-                              accept(l, ec);
-                            });
+    boost::asio::spawn(context,
+      [this, &l] (boost::asio::yield_context yield) mutable {
+        accept(l, yield);
+      }, bind_cancellation_slot(l.signal.slot(),
+             bind_executor(context, boost::asio::detached)));
+
   }
 
   ldout(ctx(), 4) << "frontend unpaused" << dendl;
 }
 
 } // anonymous namespace
 
+void RGWAsioBackoff::update_wait_time()
+{
+  if (cur_wait < max_msecs) {
+    cur_wait = cur_wait << 1;
+  }
+  if (cur_wait > max_msecs) {
+    cur_wait = max_msecs;
+  }
+}
+
+void RGWAsioBackoff::backoff_sleep()
+{
+  update_wait_time();
+  std::this_thread::sleep_for(std::chrono::milliseconds(cur_wait));
```

Comment:
the reason for coroutines in the frontend is so that we can wait without blocking the calling thread. let's use one of asio's async timers here. there's a `class basic_timeout_timer` example in rgw_asio_frontend_timer.h but that's more complicated than you need here:
```c++
class RGWAsioBackoff {
  using Clock = ceph::coarse_mono_clock;
  using Timer = boost::asio::basic_waitable_timer<Clock>;
  Timer timer;
```
```c++
void RGWAsioBackoff::backoff_sleep(boost::asio::yield_context yield)
{
  update_wait_time();
  timer.expires_after(std::chrono::milliseconds(cur_wait));
  timer.async_async(yield);
}
```